DOC-11283 Feedback on Server Group Awareness | Couchbase Docs

RayOffiah opened the pull request:
https://issues.couchbase.com/browse/DOC-11283

Changed pages:

https://rayoffiah.github.io/DOC-11283/server/current/learn/clusters-and-availability/automatic-failover.html
https://rayoffiah.github.io/DOC-11283/server/current/learn/clusters-and-availability/groups.html#understanding-server-group-awareness

BKaneAtWork commented:
Ray, on the Auto-failover page, I see the note/box that you've added in the Service-Specific Auto-Failover Policy section.  It feels out of place in that section, and I think it would make more sense to move it up the prior section, Auto-Failover Constraints, possible replacing the existing comment at the end of the section referring to the Server Group Awareness page.

I like the new line about "three or more server groups" on the Groups page.  I also think that new note on the Auto-failover page could be added to the Understanding Server Group Awareness section on the Groups doc page you modified.   

On the Groups page, I would still like to see examples with 3 server groups, especially in the later example in the Group Failover and Service Availability section.  

Thanks for getting my feedback, and hopefully you'll also solicit review from our PM team.

BKaneAtWork reviewed (changes requested):
Ray, on the Auto-failover page, I see the note/box that you've added in the Service-Specific Auto-Failover Policy section. It feels out of place in that section, and I think it would make more sense to move it up the prior section, Auto-Failover Constraints, possible replacing the existing comment at the end of the section referring to the Server Group Awareness page.

I like the new line about "three or more server groups" on the Groups page. I also think that new note on the Auto-failover page could be added to the Understanding Server Group Awareness section on the Groups doc page you modified.

On the Groups page, I would still like to see examples with 3 server groups, especially in the later example in the Group Failover and Service Availability section.

Thanks for getting my feedback, and hopefully you'll also solicit review from our PM team.

RayOffiah commented:
@BKaneAtWork 

Hi there.

I've added in the suggested changes.

Could you suggest someone from the PM team who might also be interested in reviewing the changes?

BKaneAtWork commented:
I think there could be a couple more tweaks, Ray.  


On the Groups page, I suggest removing/changing some of what you added: 

![Groups page Screenshot ](https://github.com/couchbase/docs-server/assets/38470665/7a382cd1-5fee-4249-a0cc-abcd6bef5367)


On the auto-failover page, I suggest moving the note under the current "see server group awareness..." line and simplifying the note itself: 

![Auto-failover page Screenshot](https://github.com/couchbase/docs-server/assets/38470665/2e2ed675-16a1-4f21-bf34-74c016ba4e39)


I still think the example in the Group Failover and Service Availability section needs to mention 3 racks and groups vs the current 2.  But that may be a good question for PM, and you can check with Hyun-Ju or Chin for their input.  

Thanks!

BKaneAtWork commented:
@RayOffiah where can we view the changed web pages for your recent commit (https://github.com/couchbase/docs-server/commit/56aaea24ca6787085f6adc31a817cc7d281d4929)?  The rayoffiah.github.io links in the comment at the top of this pull request don't seem to show the changes.  

Reminder to make the change from "an auto-failover" to "a group failover" on the auto-failover page, too.  (See image here.) 

![Screenshot 2023-08-17 at 8 34 42 AM](https://github.com/couchbase/docs-server/assets/38470665/c71403c3-e247-4881-84b3-4456baaf0738)


I saw the new image that went from 2 to 3 groups: 
![Screenshot 2023-08-17 at 8 40 42 AM](https://github.com/couchbase/docs-server/assets/38470665/745aeb6a-e54a-4f01-8553-ded00170029e)

I don't think this is a correct representation.  Maybe we're trying to do too much on this one DOC ticket.  If you want to revert that image change, we can finish this PR for DOC-11283 to mention the need for 3 groups to have auto group failover, and then raise another DOC ticket to address changing some/all of the server group awareness page examples from 2 groups to 3 groups (to be worked in cooperation with the PM team).  What do you think?

Thanks!

RayOffiah committed:
DOC-11283 Feedback on Server Group Awareness | Couchbase Docs Corrected following review
DOC-11283 Feedback on Server Group Awareness | Couchbase Docs - Correcting changes following review
DOC-11283 Feedback on Server Group Awareness | Couchbase Docs - Added extra note to clarify that auto failover will only occur if the auto-failover conditions are met.
DOC-11283 Feedback on Server Group Awareness | Couchbase Docs - Corrected formatting
DOC-11283 Feedback on Server Group Awareness | Couchbase Docs - Rolled back most of the changes following the review, and moved details about failover for 7.1+ to the server awareness section.

RayOffiah commented:
@BKaneAtWork 

The changes are here:

https://rayoffiah.github.io/DOC-11283/server/current/learn/clusters-and-availability/groups.html

hyunjuV commented on modules/learn/pages/clusters-and-availability/groups.adoc:
```diff
@@ -15,8 +15,18 @@ Specifically, it protects a cluster from large-scale infrastructure failure, thr
 
-A server group can be automatically _failed over_: thus, if the entire group goes offline, and active vBuckets and indexes are thereby inaccessible, the replica vBuckets and replica indexes that remain available on another group can be automatically promoted to active status.
+A server group can be automatically _failed over_: thus, if the entire group goes offline, and active vBuckets and indexes are thereby inaccessible, then the replica vBuckets and replica indexes that remain available on another group can be automatically promoted to active status.
+
```
This is identical to the original sentence except for the "then" -- a typo?

RayOffiah replied:
Correction. Not a typo.

hyunjuV commented on modules/learn/pages/clusters-and-availability/groups.adoc:
```diff
@@ -15,8 +15,18 @@ Specifically, it protects a cluster from large-scale infrastructure failure, thr
+
+See xref:learn:clusters-and-availability/automatic-failover.adoc[Automatic Failover].
 In 7.1+, this is achieved by setting the maximum count for auto-failover to a value equal to or greater than the number of nodes in the server group.
+
```
The sentence in line 23 is not needed since the information is repeated in the paragraph following -- should remove this sentence in line 23:
"In 7.1+, this is achieved by setting the maximum count for auto-failover to a value equal to or greater than the number of nodes in the server group."

hyunjuV commented on modules/learn/pages/clusters-and-availability/groups.adoc:
```diff
@@ -15,8 +15,18 @@ Specifically, it protects a cluster from large-scale infrastructure failure, thr
 In 7.1+, this is achieved by setting the maximum count for auto-failover to a value equal to or greater than the number of nodes in the server group.
+
+Note that In 7.1+, _automatic failover_ can fail over more than three nodes concurrently: this has permitted the removal of pre-7.1 interfaces that were specific to triggering _auto-failover for server groups_.
+Consequently, in order to ensure successful auto-failover of a server group, the maximum count for auto-failover must be established by the administrator as a value equal to or greater than the number of nodes in the server group.
```
1. The sentence in line 26 should be changed to **not** use the phrase "in order to ensure successful auto-failover" -- instead, use the phrase "for auto-failover of a server group to be possible":

Consequently, for auto-failover of a server group to be possible, the maximum count for auto-failover must be established by the administrator as a value equal to or greater than the number of nodes in the server group.  

2. And, the following additional info should be included for clarity and emphasis in the same paragraph:

Note that for auto-failover to occur, all of the auto-failover constraints listed in xref:./automatic-failover.adoc#auto-failover-constraints[Auto-failover Constraints] must be met, including the majority quorum requirement -- that the remaining nodes must be able to form a majority quorum to be able to initiate an auto-failover.  If, for example, you have two server groups, with equal number of nodes in each, and if all of the nodes in one of the server groups fail at the same time, even if you have the maximum count for auto-failover set to a value equal to or greater than the number of nodes in the server group that failed, auto-failover cannot occur since the remaining nodes in the remaining server group cannot form a majority quorum.

hyunjuV commented on modules/learn/pages/clusters-and-availability/groups.adoc:
```diff
@@ -15,8 +15,18 @@ Specifically, it protects a cluster from large-scale infrastructure failure, thr
+Note that In 7.1+, _automatic failover_ can fail over more than three nodes concurrently: this has permitted the removal of pre-7.1 interfaces that were specific to triggering _auto-failover for server groups_.
+Consequently, in order to ensure successful auto-failover of a server group, the maximum count for auto-failover must be established by the administrator as a value equal to or greater than the number of nodes in the server group.
+
+Note also that in pre-7.1, interfaces for triggering auto-failover for server groups have been _removed_ from 7.1: therefore, programs that attempt to use the pre-7.1 interfaces with 7.1+ will _fail_. +
```
This line -- line 28 -- seems a bit out-of-place here -- best to remove since the previous paragraph already noted the removal of pre-7.1 interfaces.

hyunjuV commented on modules/learn/pages/clusters-and-availability/groups.adoc:
```diff
@@ -201,9 +212,4 @@ To enable the failover of all nodes in a group:
-Note that in 7.1+, _automatic failover_ can fail over more than three nodes concurrently: this has permitted the removal of pre-7.1 interfaces that were specific to triggering _auto-failover for server groups_.
-Consequently, in order to ensure successful auto-failover of a server group, the maximum count for auto-failover must be established by the administrator as a value equal to or greater than the number of nodes in the server group.
 
-Note that the pre-7.1 interfaces for triggering auto-failover for server groups have been _removed_ from 7.1: therefore, programs that attempt to use the pre-7.1 interfaces with 7.1+ will _fail_.
```
Lines 207-209 should be kept.   I realize I had previously said that they should be removed, but looking at the information in this section again, I see that it's needed.  Also, users may navigate to this section from the right-hand menu without looking at the rest of the page.  So, this info is needed and appropriate here.

hyunjuV commented on modules/learn/pages/clusters-and-availability/groups.adoc:
```diff
@@ -201,9 +212,4 @@ To enable the failover of all nodes in a group:
 * With CLI, see xref:cli:cbcli/couchbase-cli-setting-autofailover.adoc[setting-autofailover].
 * With the REST API, see xref:rest-api:rest-cluster-autofailover-enable.adoc[Enabling and Disabling Auto-Failover].
 
-Note that in 7.1+, _automatic failover_ can fail over more than three nodes concurrently: this has permitted the removal of pre-7.1 interfaces that were specific to triggering _auto-failover for server groups_.
```
We need a modified version of this paragraph in lines 204-205. I realize I had previously said that they should be removed, but looking at the information in this section again, I see that it's needed.  Also, users may navigate to this section from the right-hand menu without looking at the rest of the page.  

The paragraph should be the same one from the top of the page:

Note that In 7.1+, _automatic failover_ can fail over more than three nodes concurrently: this has permitted the removal of pre-7.1 interfaces that were specific to triggering _auto-failover for server groups_. 
Consequently, for auto-failover of a server group to be possible, the maximum count for auto-failover must be established by the administrator as a value equal to or greater than the number of nodes in the server group.  Note that for auto-failover to occur, all of the auto-failover constraints listed in xref:./automatic-failover.adoc#auto-failover-constraints[Auto-failover Constraints] must be met, including the majority quorum requirement -- that the remaining nodes must be able to form a majority quorum to be able to initiate an auto-failover. If, for example, you have two server groups, with equal number of nodes in each, and if all of the nodes in one of the server groups fail at the same time, even if you have the maximum count for auto-failover set to a value equal to or greater than the number of nodes in the server group that failed, auto-failover cannot occur since the remaining nodes in the remaining server group cannot form a majority quorum.

BKaneAtWork commented on modules/learn/pages/clusters-and-availability/groups.adoc:
```diff
@@ -15,8 +15,32 @@ Specifically, it protects a cluster from large-scale infrastructure failure, thr
+
+See xref:learn:clusters-and-availability/automatic-failover.adoc[Automatic Failover].
+
+Note that In 7.1+, automatic failover can fail over more than three nodes concurrently: this has permitted the removal of pre-7.1 interfaces that were specific to triggering auto-failover for server groups.
```
change "In" to "in" (lowercase)

RayOffiah replied:
done

BKaneAtWork commented on modules/learn/pages/clusters-and-availability/groups.adoc:
```diff
@@ -15,8 +15,32 @@ Specifically, it protects a cluster from large-scale infrastructure failure, thr
+
+NOTE: For the vBuckets and replica indexes to be automatically promoted to active, the conditions specified in xref:./automatic-failover.adoc#auto-failover-constraints[Auto-failover Constraints] must apply.
+
+See xref:learn:clusters-and-availability/automatic-failover.adoc[Automatic Failover].
```
this link on line 22 seems redundant since the link is also on line 20 and line 44.

BKaneAtWork commented on modules/learn/pages/clusters-and-availability/groups.adoc:
```diff
@@ -15,8 +15,32 @@ Specifically, it protects a cluster from large-scale infrastructure failure, thr
+====
+
+
+Note also that in pre-7.1, interfaces for triggering auto-failover for server groups have been _removed_ from 7.1: therefore, programs that attempt to use the pre-7.1 interfaces with 7.1+ will _fail_. +
```
Seems redundant given the info is (mostly) in line 24.  Maybe the text following the colon ("therefore...") should be moved to line 24 if needed.  If there's a reason to keep line 42, re-word the awkward beginning "Note also that in pre-7.1, interfaces..." to something like "Note also that pre-7.1 interfaces..."

RayOffiah replied:
I've moved the text closer to the note about 7.1.

BKaneAtWork reviewed (changes requested):
A few small changes could be made, per my 3 comments.

BKaneAtWork reviewed (approved):
Looks good, thanks @RayOffiah
